packages/sign-client/test/sdk/integration/integration.spec.ts

```diff
@@ -0,0 +1,118 @@
+import { getSdkError } from "@walletconnect/utils";
+import { expect, describe, it, vi, afterAll, beforeAll } from "vitest";
+import { createExpiringPromise } from "../../../../utils/src";
+import { initTwoClients, testConnectMethod, deleteClients, Clients } from "../../shared";
+const TESTS_CONNECT_RETRIES = 5;
+const TESTS_CONNECT_TIMEOUT = 20_000;
+describe("Sign Client Integration", () => {
+  let clients: Clients;
+  let pairingA: any;
+  let sessionA: any;
+
+  beforeAll(async () => {
+    clients = await initTwoClients();
+    let retries = 0;
+    while (!pairingA) {
+      if (retries > TESTS_CONNECT_RETRIES) {
+        throw new Error("Could not pair clients");
+      }
+      try {
+        const settled: any = await createExpiringPromise(
+          testConnectMethod(clients),
+          TESTS_CONNECT_TIMEOUT,
+        );
+        pairingA = settled.pairingA;
+        sessionA = settled.sessionA;
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.log("retrying", e);
+      }
+      retries++;
+    }
+  });
+
+  afterAll(async () => {
+    await deleteClients(clients);
+  });
+
+  it("init", () => {
+    expect(clients.A).to.be.exist;
+    expect(clients.B).to.be.exist;
+  });
+  describe("connect", () => {
+    it("connect (with old pairing)", async () => {
+      const { A, B } = clients;
+      expect(A.pairing.keys).to.eql(B.pairing.keys);
+      await testConnectMethod(clients, {
+        pairingTopic: pairingA.topic,
+      });
+    });
+  });
+  describe("update", () => {
+    it("updates session namespaces state with provided namespaces", async () => {
+      const topic = sessionA.topic;
+      const namespacesBefore = clients.A.session.get(topic).namespaces;
+      const namespacesAfter = {
+        ...namespacesBefore,
+        eip9001: {
+          accounts: ["eip9001:1:0x000000000000000000000000000000000000dead"],
+          methods: ["eth_sendTransaction"],
+          events: ["accountsChanged"],
+        },
+      };
+      const { acknowledged } = await clients.A.update({
+        topic,
+        namespaces: namespacesAfter,
+      });
+      await acknowledged();
+      const result = clients.A.session.get(topic).namespaces;
+      expect(result).to.eql(namespacesAfter);
+    });
+  });
+
+  describe("extend", () => {
+    it.skip("updates session expiry state", async () => {
```

Comment:
Skipping here but I'm guessing this should be re-enabled if it was specifically moved to `integration.spec.ts`